### PR TITLE
PS-3871 (5.5): Typo prevents MYSQL_SERVER to be defined for statically linked plugins

### PR DESCRIPTION
**Pull request by inikep**
The MYSQL_ADD_PLUGIN macro has a spelling error causing the MYSQL_SERVER
define not to be defined. This causes the storage engine plugins to be
compiled without MYSQL_SERVER defined and thus may get a different set of
function signatures, functions and symbols unless the source of the plugins
themself define the MYSQL_SERVER define.

### CODE REVIEW

File: storage/heap/CMakeLists.txt [hunk 1]
```diff
@@ -26,8 +26,12 @@ SET(HEAP_SOURCES  _check.c _rectest.c hp_block.c hp_clear.c hp_close.c hp_create
 MYSQL_ADD_PLUGIN(heap ${HEAP_SOURCES} STORAGE_ENGINE MANDATORY
                  RECOMPILE_FOR_EMBEDDED DTRACE_INSTRUMENTED)
 
-ADD_EXECUTABLE(hp_test1 hp_test1.c)
-TARGET_LINK_LIBRARIES(hp_test1 mysys heap dbug strings)
+IF(WITH_UNIT_TESTS AND WITH_EMBEDDED_SERVER)
```
Comment (laurynas-biveinis) [inline storage/heap/CMakeLists.txt]: Does this mean that hp_tests will not get built in case of -DWITH_EMBEDDED_SERVER=OFF ?

Comment (inikep) [reply storage/heap/CMakeLists.txt]: Yes, this is backported from 5.7

**Comment by inikep** [comment]
https://jenkins.percona.com/job/percona-server-5.5-param/1637/

File: storage/heap/CMakeLists.txt (hunk 1 above)
Comment (laurynas-biveinis) [reply storage/heap/CMakeLists.txt]: Consider reporting upstream (although it might have been already fixed due to 8.0 removing embedded server)

**Review by laurynas-biveinis — APPROVED** [review]

**Comment by inikep** [comment]
PR was updated to have a minimal set of changes backported from 5.7 required to fix `COMPILE_DEFINITIONS` that is only `storage/innobase/CMakeLists.txt`

**Comment by inikep** [comment]
https://jenkins.percona.com/job/percona-server-5.5-param/1638/

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**